screen/wallets/entropyGenerator.js

```diff
@@ -0,0 +1,329 @@
+  const [selectedWalletType, setSelectedWalletType] = useState(selectedWalletTypeProps || false);
+
+  const [entropyButtonText, setEntropyButtonText] = useState(loc.wallets.add_entropy_provide);
+  const stylesHook = {
+    advancedText: {
+      color: colors.feeText,
+    },
+    label: {
+      borderColor: colors.formBorder,
+      borderBottomColor: colors.formBorder,
+      backgroundColor: colors.inputBackgroundColor,
+    },
+    noPadding: {
+      backgroundColor: colors.elevated,
+    },
+    root: {
+      backgroundColor: colors.elevated,
+      marginBottom: 30,
+    },
+    lndUri: {
+      borderColor: colors.formBorder,
+      borderBottomColor: colors.formBorder,
+      backgroundColor: colors.inputBackgroundColor,
+    },
+  };
+
+  useEffect(() => {
+    isTestModeEnabled()
+      .then(setIsTestMode)
+      .finally(() => setIsLoading(false));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const generateEntropyDrawing = async () => {
+    const flatArray = paths.flatMap((path) => path.segments);
+    const numArr = flatArray.flatMap((str) => {
+      // Split the string into parts by spaces
+      const parts = str.split(' ');
+      // Convert the numerical parts into numbers
+      const nums = parts.slice(1).map(parseFloat);
+      // Return an array with the first part and the numbers
+      return nums;
+    });
+    // The following function takes a user-provided “randomness” value and XORs it with a PRNG-generated value to improve the randomness of the output.
+    const normalizedPoints = normalize(numArr.map((point) => Math.round(point)));
+    return generateEntropy(normalizedPoints);
+  };
+
+  const createWallet = async () => {
+    setIsLoading(true);
+
+    let w;
+
+    if (selectedWalletType === ButtonSelected.ONCHAIN) {
+      const entropy = await generateEntropyDrawing();
+      const shuffledEntropy = getNRandomElementsFromArray(entropy, 16);
+      if (entropy.length < 192) {
+        alert('Not enough entropy gathered. Please continue drawing.');
+        setIsLoading(false);
+        return;
+      }
+      if (selectedIndex === 2) {
+        // zero index radio - HD segwit
+        w = new HDSegwitP2SHWallet();
+        w.setLabel(label || loc.wallets.details_title);
+      } else if (selectedIndex === 1) {
+        // btc was selected
+        // index 1 radio - segwit single address
+        w = new SegwitP2SHWallet();
+        w.setLabel(label || loc.wallets.details_title);
+      } else {
+        // btc was selected
+        // index 2 radio - hd bip84
+        if (isTestMode) {
+          w = new HDSegwitBech32Wallet({ network: bitcoin.networks.testnet });
+        } else {
+          w = new HDSegwitBech32Wallet();
+        }
+        w.setLabel(label || loc.wallets.details_title);
+      }
+      if (selectedWalletType === ButtonSelected.ONCHAIN) {
+        if (selectedIndex === 1) {
+          // segwit (p2sh) without Entropy
```

Comment:
? I'm not sure I understand this comment

Reply:
@muursh  If a user wants to create segwit p2sh wallet so it will be without entropy ,  

OR if we have decide about this sewgit p2sh  so I can remove this condition because this code will move to add.js file .